ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/KsqlMaterialization.java

```diff
@@ -185,16 +185,38 @@ public MaterializedWindowedTable windowed() {
     public Iterator<WindowedRow> get(final int partition, final Range<Instant> windowStartBounds,
         final Range<Instant> windowEndBounds) {
       final Iterator<WindowedRow> result = table.get(partition, windowStartBounds, windowEndBounds);
-
       return Streams.stream(result)
           .map(row ->  {
-            return filterAndTransform(row.windowedKey(), row.value(), row.rowTime())
+            return filterAndTransform(row.windowedKey(), getIntermediateRow(row), row.rowTime())
                 .map(v -> row.withValue(v, schema()));
           })
           .filter(Optional::isPresent)
           .map(Optional::get)
           .iterator();
     }
   }
+
+  public static GenericRow getIntermediateRow(final TableRow row) {
```

Comment:
It would be nice to just call https://github.com/confluentinc/ksql/blob/master/ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/operators/PullPhysicalOperatorUtil.java since this is effectively part of the physical layer for pull queries.